ID.me WebVerify SDK/IDmeWebVerify.m

```diff
@@ -95,91 +97,126 @@ - (void)verifyUserInViewController:(UIViewController *)externalViewController
-                } else {
-                    NSError *modifiedError = [[NSError alloc] initWithDomain:IDME_WEB_VERIFY_ERROR_DOMAIN
-                                                                        code:IDmeWebVerifyErrorCodeVerificationDidFailToFetchUserProfile
-                                                                    userInfo:error.userInfo];
-                    webVerificationResults(nil, modifiedError, nil);
-                }
-
-                // Dismiss _webViewController and clear _webView cache
-                if (self.webNavigationController) {
-                    [self destroyWebNavigationController:self];
-                }
-            });
-            
-        }];
-        [task resume];
+
+                                if (!accessToken) {
+                                    NSError *error = [[NSError alloc] initWithDomain:IDME_WEB_VERIFY_ERROR_DOMAIN
+                                                                                code:IDmeWebVerifyErrorCodeNoSuchScope
+                                                                            userInfo:nil];
+                                    webVerificationResults(nil, error, nil);
+                                    // Dismiss _webViewController and clear _webView cache
+                                    if (self.webNavigationController) {
+                                        [self destroyWebNavigationController:self];
+                                    }
+                                    return;
+                                }
+                                NSString *requestString = [NSString stringWithFormat:IDME_WEB_VERIFY_GET_USER_PROFILE, accessToken];
+
+                                requestString = [requestString stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding];
+                                NSURL *requestURL = [NSURL URLWithString:requestString];
+                                NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:requestURL];
+                                [request setValue:@"IDmeWebVerify-SDK-iOS" forHTTPHeaderField:@"X-API-ORIGIN"];
+
+                                NSURLSession *session = [NSURLSession sharedSession];
+                                NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *error) {
+                                    dispatch_async(dispatch_get_main_queue(), ^{
+                                        NSHTTPURLResponse* httpResponse = (NSHTTPURLResponse*)response;
+                                        NSUInteger statusCode = [httpResponse statusCode];
+                                        if ([data length]) {
+                                            NSError* error;
+                                            NSDictionary *results = (NSDictionary *)[NSJSONSerialization JSONObjectWithData:data options:NSJSONReadingAllowFragments error:&error];
```

Comment:
this is only needed if `statusCode == 200`, right? maybe you can move it to that `if`.

--------

I'm not sure about the structure of this `if else` block. What would happen if `statusCode == 401` and the response doesn't have a body? Currently it is returning the error `IDmeWebVerifyErrorCodeVerificationDidFailToFetchUserProfile` instead of doing the refresh process.

What do you think?